### PR TITLE
WIP: MacOS Intel nightly build using Macports

### PR DESCRIPTION
**Pull request by jonastr**
NOTE: this is a work in progress. The build does not yet succeed. I am opening this PR already to get your feedback on the general idea.
I'll put some comments on changes that I only made to get quick feedback on my own fork (https://github.com/jonastr/darktable/pull/2). These would of course be reverted+squashed before merge.

Current problem - if anyone has an idea, please comment: Getting CMake to find OpenMP. I've checked the [CMake find scripts](https://cmake.org/cmake/help/latest/module/FindOpenMP.html), but they do not seem to support any find hints.

I am aiming for one nightly Mac build targeting Intel platform for now. Right now I have the matrix build enabled so that I get faster feedback on which combination succeeds the build.

While a homebrew-based build did already succeed (check this PR on my own fork: https://github.com/jonastr/darktable/pull/1), I am trying to get Macports to run as this would be in-line with the existing Mac bundle for the release builds. Otherwise, two bundler scripts + configuraiton would need to be maintained.

### CODE REVIEW

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -4,10 +4,13 @@ on:
   schedule:
     - cron: "0 0 * * *"
   workflow_dispatch:
+  push:
```
Comment (jonastr) [inline .github/workflows/nightly.yml:7]: only to get quick feedback independent of nightly cron

**Comment by jonastr** [comment]
@parafin do you have any idea where this build error is coming from?

https://github.com/jonastr/darktable/pull/2/checks?check_run_id=3914722092

```
/Users/runner/work/darktable/darktable/src/src/common/imageio_rawspeed.cc:133:15: fatal error: no matching constructor for initialization of 'rawspeed::RawParser'
    RawParser t(*m.get());
              ^ ~~~~~~~~
/Users/runner/work/darktable/darktable/src/src/external/rawspeed/src/librawspeed/parsers/RawParser.h:35:12: note: candidate constructor not viable: no known conversion from 'const rawspeed::Buffer' to 'const rawspeed::Buffer *' for 1st argument; remove *
  explicit RawParser(const Buffer* inputData) : mInput(inputData) {}
           ^
/Users/runner/work/darktable/darktable/src/src/external/rawspeed/src/librawspeed/parsers/RawParser.h:33:7: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'const rawspeed::Buffer' to 'const rawspeed::RawParser' for 1st argument
class RawParser {
      ^
```

i do remember that i had this also on my local machine with M1, but got it resolved somehow. unfortunately, don't remember anymore what i did. :)

**Comment by parafin** [comment]
Your PR changes rawspeed submodule, while it shouldn’t. Remove that change from commits.

**Comment by jonastr** [comment]
i am closing this in favor of a homebrew-based build as GH runners come with homebrew pre-installed (and uninstalling is not an option, it breaks the runners themselves). See https://github.com/darktable-org/darktable/pull/10360.

**Pull request closed without merging**